File: python/src/citronella/logger.py
from logging import info


def logger(f):
    """This is a logger wrapper."""
    def func(*args, **kwargs):
        """This is a logger method."""
        if args[0]._logger:
            info(f'{args[0]._class_name} => '
                 f'{args[0]._function_name} => {f.__name__}')
        return f(*args, **kwargs)

    return func

File: ruby/lib/logger.rb
require 'logger'

module Citronella
  module Log
    def self.logger(logger, class_name, function_name, name)
      if logger
        Logger.new(STDOUT, level: Logger::INFO).info("#{class_name} => #{function_name} => #{name}")
      end
    end
  end
end
